spynnaker/pyNN/models/neuron/master_pop_table.py

```diff
@@ -189,32 +227,74 @@ def initialise_table(self):
         self.__n_single_entries = 0
 
     def update_master_population_table(
-            self, block_start_addr, row_length, key_and_mask, is_single=False):
+            self, block_start_addr, row_length, key_and_mask, core_mask,
+            core_shift, n_neurons, is_single=False):
         """ Add an entry in the binary search to deal with the synaptic matrix
 
-        :param spec: the writer for DSG
         :param block_start_addr: where the synaptic matrix block starts
         :param row_length: how long in bytes each synaptic entry is
         :param key_and_mask: the key and mask for this master pop entry
         :type key_and_mask: \
             :py:class:`pacman.model.routing_info.BaseKeyAndMask`
-        :param master_pop_table_region: the region ID for the master pop
+        :param core_mask: Mask for the part of the key that identifies the core
+        :param core_shift: The shift of the mask to get to the core_mask
+        :param n_neurons: \
+            The number of neurons in each machine vertex (bar the last)
         :param is_single: \
             Flag that states if the entry is a direct entry for a single row.
         :return: The index of the entry, to be used to retrieve it
         :rtype: int
+        :raises SynapticConfigurationException: If a bad address is used.
         """
         # pylint: disable=too-many-arguments, arguments-differ
         if key_and_mask.key not in self.__entries:
             self.__entries[key_and_mask.key] = _MasterPopEntry(
-                key_and_mask.key, key_and_mask.mask)
-        start_addr = block_start_addr
+                key_and_mask.key, key_and_mask.mask, core_mask, core_shift,
+                n_neurons)
 
-        # if single, don' t add to start address as its going in its own block
+        # if not single, scale the address
+        start_addr = block_start_addr
         if not is_single:
-            start_addr = block_start_addr // self.ADDRESS_SCALE
+            if block_start_addr % _ADDRESS_SCALE != 0:
+                raise SynapticConfigurationException(
+                    "Address {} is not compatible with this table".format(
+                        block_start_addr))
+            start_addr = block_start_addr // _ADDRESS_SCALE
+            if start_addr & _ADDRESS_MASK != start_addr:
+                raise SynapticConfigurationException(
+                    "Address {} is too big for this table".format(
+                        block_start_addr))
+        if (row_length - 1) & _ROW_LENGTH_MASK != (row_length - 1):
```

Comment:
this test is located in lines 198 to 206. replace. but the same comment about row_length -1 stands there too

Reply:
Fixed